File: Python/7.py
# Reverse Integer

class Solution:
    def reverse(self, x: int) -> int:
        if x == 0:
            return 0

        string_number: str

        if x > 0:
            string_number = str(x)[::-1]
        else:
            string_number = str(x).removeprefix('-')
            string_number = string_number[::-1]

        new_number: int = int(string_number)

        if new_number > 2**31 - 1 or new_number < -2**31:
            return 0

        if x < 0:
            return new_number * -1

        return new_number

File: JS/7.js
// Reverse Integer

var reverse = function(x) {
    if(x === 0) return 0;

    if(x/10 === 0) return x;

    let temp;
    let isFirstNumber = true;
    let final = '';
    let isNegative = false;
    let newNumber;

    if(x < 0){
        x *= -1;
        isNegative = true;
    }

    while(x > 0){
        temp = x % 10;
        x /= 10;
        if (isFirstNumber && temp === 0){
            isFirstNumber = false;
            continue;
        }
        final += temp.toString();
        if (isFirstNumber) isFirstNumber = false;
    }

    newNumber = Number.parseInt(final);

    if (isNegative) newNumber = -newNumber;

    if (newNumber > Number.MAX_SAFE_INTEGER || newNumber < Number.MAX_SAFE_INTEGER ) return 0;

    return newNumber;
};